test/runtime-puppeteer/samples/inline-style-directive-important/main.svelte

```diff
@@ -0,0 +1,13 @@
+<script>
+	export let color = 'red'
+</script>
+
+<h1 style:background-color={color} >hello</h1>
+<h1 style:background-color="{color} important">hello</h1>
```

Comment:
Is `!important` more accurate? (Or should we check both?)

```suggestion
<h1 style:background-color="{color} !important">hello</h1>
```

And then, in this case, only SSR'd HTML will apply the red background color.
I think we should have the same behavior between DOM and SSR.

Reply:
That one can probably be removed actually. I think Li Hau added it to test parsing for false positives in the regex based impl, but that's not needed in the modifier impl.

Reply:
What do you think about it?🤔
Should we introduce RegExp way for SSR?

> And then, in this case, only SSR'd HTML will apply the red background color.
I think we should have the same behavior between DOM and SSR.

Reply:
if `style:background-color|important={color}` is the way to go, i feel like we can have some runtime dev warnings against having passed in `"red !important"` as value in the style directives, and nudge them to use the `|important` modifier instead.

because it is possible to have the following, and not able to detect important from regex.

```
<script>
   let color = "red !important";
</script>

<h1 style:background-color={color} />
```

this renders red color in SSR and no color in DOM too

we actually have the same problem with normal style attribute too

```
<script>
	let color = 'red !important';
</script>

<h1 style="color: {color}">hello</h1>
```

Reply:
Yes, we can not handle this at compile time but we can handle this out at SSR runtime, right?

But I agree that we will fix this issue on another PR because as you said that this issue comes from an existing issue.

Reply:
Now that I think about it, we can implement this in runtime with a tiny bit of additional code, SSR runtime  and emitted code can remain unchanged

In the dom's set_style, which was the source of error in the original issue:
```js
function set_style(node, property, value) {
    let priority;
    node.style.setProperty(
        property,
        value.replace(/\s*!important$/, () => (priority = "important", "")),
        priority
    )
}
```
this will slightly increase the cost of set_style, although not sure how significant it would be

Reply:
I think we don't need two ways to specify important, and since the compiler's purpose is to reduce runtime work, I think it's best to avoid adding runtime work here as well.

But I would like to hear others' opinions.

Reply:
yes i agree.

we should only support `<div style:color|important="red">`

and if you are passing `<div style:color="red !important">` it is not going to work. and in dev runtime, you will see a warning telling you that you should use `style:color|important` instead.

Reply:
i've removed this extra test